notebooks/Notebook1.md

```diff
@@ -2,33 +2,51 @@
+
+
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+I started by focusing in on room navigation. What I mean by this is when you are in the game, being able to say 'go south' or 'go west' and be able to navigate to that adjacent room when there is one present. I then sketched out the DSL design for room declaration, text, and navigation, where I settled on an HTML-like structure with directions specified in clear English. See the picture of my design below:
+
+![My first take](pictures/DSLs-3.jpg)
+
+Once the initial DSL was solidified, I implemented the backend to make it all work. I built an Engine class that has the game loop (while True) and parses user input for expected commands. I also wrote a Room class to handle each room, which right now includes a room ID, room text (that gets printed when you enter it) and a 4-member array of ints that connects to the room IDs of adjacent rooms. 
+
+My last task this week was to write up a game with just connected rooms in my DSL (done in a file named adventure.game), then implement the parsing and linking with my backend. After some searching, I decided to use the pyparsing package because it is robust and well-documented. Using that package I was able to succesfully parse my external DSL code and do some additional processing to interface with my backend.
+
+So now my basic game works! You can run 'python3 parseGame.py adventure.game' and it will parse the game from the raw DSL syntax and start the engine so you can play it in the terminal. Feel free to do that if you clone my artifact repo (and poke around at the underlying code if you wish).
+
+
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+Now that room movement works well from front to back, I think the most pressing issue is looking ahead to more complex objects in the game, such as items and characters. I need to narrow down the exact scope for what you can do with an item in my language. For example, can items be combined together to form something new? Additionally, what needs to be an object in the space? Should a locked door be considered an item, or should the room class be revised to support that? I'll have to think a lot about this over the course of the next week!
```

Comment:
Personally, I think it makes the most sense for a locked door to be considered a property of the room instead of an item, but I see why you would be conflicted, though. A locked door might have properties more befitting of an item; for example, what if you want to check the locked door? In some RPGs, you might get the message "It's locked" upon checking it, so perhaps you also want to have that. However, unlike most items, you can't carry or drop a door. 

You could alternatively make a door class. Suppose that each room can be visualized as a node in a graph; then, the doors might be considered the edges between the nodes. If you view it this way, doors and rooms seem different enough that it warrants having a separate class. However, the drawback is this: suppose you have room A and room B with a connecting door AB. Do you want the door to have the same description no matter what room you check it from? What if there's a riddle of some sort posted on only one side of the door? This is probably a rare edge case, but it's worth considering.